umbra-js/src/classes/Umbra.ts

```diff
@@ -759,8 +770,10 @@ async function recursiveGraphFetch(
 
   // If there were results on this page then query the next page, otherwise return the data
   /* eslint-disable @typescript-eslint/no-unsafe-return */
-  if (!json.data[key].length) return [...before];
-  else return await recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides);
+  if (json.data[key].length) {
+    yield json.data[key]; // yield the data for this page
+    yield* recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides); // yield the data for the next pages
+  }
```

Comment:
Was the `else` block intentionally dropped here? Seems like a bug was introduced
```suggestion
  if (json.data[key].length) yield json.data[key]; // yield the data for this page
  else yield* recursiveGraphFetch(url, key, query, [...before, ...json.data[key]], overrides); // yield the data for the next pages
```